### ISSUE TITLE
WarnGen Templates?

### ISSUE THREAD

**Issue by airbusman5514** [Unidata/awips2#178]
Hello, all...

I know this probably isn't the best place to ask, but where can I download WarnGen templates? Or is there a guide on writing my own? Thanks!

WarnGen Junkie

**Comment by electricbrooke** [comment]
CentOS or Mac?

On CentOS, they can be found in caveData > common > base > warngen. You will need to go into Warngen and click on each individual warning (and generate a warning, even if you don't "send" it); then each warning's file will appear. For each warning the .vm file controls text output while the .xml file controls the actual buttons available when you load a warning in WarnGen (for example I could disable the ability to select penny and nickel size hail in a severe thunderstorm warning with the .xml). Meanwhile config.xml is your localization file, which lets you pick the WFO name, what warnings are available, etc. (As for the site name, using Taunton, MA (BOX) as an example, my config file looks like this:
```
<configuration>
<warngenOfficeShort>Taunton MA</warngenOfficeShort><warngenOfficeLoc>Taunton</warngenOfficeLoc>
<backupCWAs>GYX</backupCWAs>
<siteNode>BOX</siteNode>
<defaultTemplate>impactSevereThunderstormWarning</defaultTemplate><mainWarngenProducts>Tornado/impactTornadoWarning,Severe Thunderstorm/impactSevereThunderstormWarning,Severe Weather Statement/impactSevereWeatherStatement,Marine Warning/impactSpecialMarineWarning,Flash Flood Warning/flashFloodWarning</mainWarngenProducts>
<otherWarngenProducts>Flash Flood Statement/flashFloodWarningFollowup,Non-Convective FFW (incl. Dam Break)/nonConvectiveFlashFloodWarning,Non-Convective Flash Flood Statement/nonConvectiveFlashFloodWarningFollowup,Areal Flood Warning/arealFloodWarning,Areal Flood Warning Followup/arealFloodWarningFollowup,Areal Flood Advisory/arealFloodAdvisory,Areal Flood Advisory Followup/arealFloodAdvisoryFollowup,Special Weather Statement/specialWeatherStatement,Marine Warning Followup/impactSpecialMarineWarningFollowup,Marine Statement/marineWeatherStatement</otherWarngenProducts>
```

And yields this at the top of a warning:

"BULLETIN - IMMEDIATE BROADCAST REQUESTED
Severe Thunderstorm Warning
National Weather Service Taunton MA
858 PM EDT Wed Jul 12 2017

The National Weather Service in Taunton has issued a

* Severe Thunderstorm Warning for...
  Central Essex County in northeastern Massachusetts..." and so on. (I am mostly referencing the different references to Taunton and where they should be in the config.xml.)

I too am a Warngen addict that spent too much time last summer creating options for severe thundersnow warnings, adjusting the SVR template to add 30 mph wind gusts or pea size hail as options so long as some severe threat was also chosen, etc.

On Mac, the procedure might be similar but I have not tested it on the Mac in my house yet as it is not my main computer (rather a house computer).

**Comment by airbusman5514** [comment]
Wait, that was you sending those Taunton warnings? Cool. Didn't realize they sent out over LAN. Either way it was cool. Is there a guide to writing them?

**Comment by airbusman5514** [comment]
Another question: how do I get the TORNADO...OBSERVED, HAIL...1.00 IN, etc. tags into WarnGen?
Also, I think I screwed up my TOR template. It doesn't list times in the Pathcast mode, only a bunch of code.

**Comment by electricbrooke** [comment]
Yeah I had no idea those sent over LAN with the T code enabled. That's strange. But yeah, if it's a Taunton warning it's most likely mine -- and I sometimes go over the top with it. I used to send them operationally onto the Unidata server last summer on Windows AWIPS, but now I primarily use Linux AWIPS so (I hope) it is on testing mode like Unidata AWIPS is programmed to be. Whatever the case, not a big deal. How did you see the Taunton warnings on your display just for my future reference? (curious)

As for the tags, switch your config.xml to use the Impact-Based Warnings:
change tornadoWarning to impactTornadoWarning
change severeThunderstormWarning to impactSevereThunderstormWarning
change severeWeatherStatement to impactSevereWeatherStatement
if marine weather applies for the CWA you are localized to:
specialMarineWarning becomes impactSpecialMarineWarning
specialMarineWarningFollowup becomes impactSpecialMarineWarningFollowup.

The pathcast thing I can't help you with directly (I never figured out that part of WarnGen because I don't really use it, if I want to calculate Time of Arrival I use the actual tool for that in AWIPS) but if you are on "tornadoWarning.vm" you can just switch it to "impactTornadoWarning.vm" and get a fresh start. To reset the entire template just go back to the place where the files are and delete the .vm file, then generate another warning of that type and voila!

**Comment by airbusman5514** [comment]
Cool. Thanks. Turns out we're not the only guys using WarnGen. Type in WRKWG0 in the AFOS Cmd box and you'll see bulletins from all over.

Your warnings popped up in the text box. I'd get a header block for BOXSVRBOX and I'd be able to edit it. At first when I saw them I thought they were being issued by the real NWS office in Taunton since they didn't have the test markings.

What other warning types are available to use from WarnGen? So far I just have TOR, SVR, SVS, SPS, FFW, and that tiny list of mostly flood warnings under Other. What can I expand that with?

**Comment by electricbrooke** [comment]
That is...weird. It **shouldn't** be popping up in someone else's text box, but at least that probably explains why when I went to eat dinner at 5:50 last night I went back to my desk at 6:05 and immediately saw some warning from Indiana on my AWIPS screen. Sorry if I was causing any disruption with all the BOX warnings (including the three or four proposed TORs that I never clicked "send" on.)

Anyways, you can add Special Marine Warnings, Marine Weather Statements, among a list of others that I seem to forget. Let me find that whole list on Windows (the only OS where you can actually localize WarnGen from the localization box and where they should all be), I will edit this post when I am done.

**Comment by airbusman5514** [comment]
Yeah I just thought there was some freak weather anomaly in New England. I figured if NWS Taunton’s issuing those it should be okay, but either way, no problem. I hope the TOR from IND didn’t disrupt you. I think I’ve gotten the hang of it. I did, however, try to add Blizzard Warning to the Other list, but to no avail (I’m guessing that’s GFE).

 

Do you know if the tornado emergency I just made actually went out? I doubt it did but still...

**Comment by electricbrooke** [comment]
Well we did have actual severe weather yesterday in which the real NWS Taunton issued two areal flood warnings, seven flash flood warnings, thirteen areal flood advisories and thirteen severe thunderstorm warnings. So it wasn't totally far fetched. I only saw one severe weather statement from IND, and I didn't even see it pop up immediately because I was at the dinner table.

Referring to my prior post, Special Marine Warning is impactSpecialMarineWarning and impactSpecialMarineWarningFollowp (remove impact and change the s in "special" to lowercase if you want the old style). Non-ashfall Marine Weather Statements are marineWeatherStatement.

Anyway... All warnings not mentioned in the last post that are available in Warngen:
airportWeatherWarning (must be edited extensively for localization)
burnScarFlashFloodWarning (and burnScarFlashFloodWarningFollowup)
dambreak (though this seems to conflict with the nonConvectiveFlashFloodWarning, will have to investigate)
extremeWindWarning (and extremeWindWarningFollowup)
fireWarning 
marineWeatherStatementAshfall
shortTermForecast (though this might already be in there)

Blizzard Warnings and the like are in GFE, and for some reason, GFE refuses to load on CentOS 7 for me. :( 
(I haven't tested it on Windows or Mac.)

**Comment by airbusman5514** [comment]
Cool. Sorry for the sudden influx of questions, but how do you remove the test wording from the WarnGen templates? I tried yesterday and it wouldn’t go. I also couldn’t get PathCast to load any times. It would just say “remain over mainly rural areas of Marion County” even though Indianapolis and most of its suburbs are in Marion County, hardly rural in my opinion.

**Comment by electricbrooke** [comment]
Test wording: Go into the .vm file and find all instances of "#if(${productClass}=="T")" then remove the test wording - but don't remove the whole block. If ## is in there, leave them. If that isn't there, just leave the line blank. One whole line blank. As for the top of each warning, just remove the "TEST..." from each side (unless you want to keep it). The Pathcast thing is odd.

**Comment by airbusman5514** [comment]
Have you successfully gotten highway mile markers into your bulletins?

**Comment by electricbrooke** [comment]
@airbusman5514 I have not tried to get highway mile markers in the bulletins yet (mostly because we use sequential exits in New England). But I think there's a way. I'm a little tied up right now but I will take a look later.
